core/src/main/java/hudson/scm/SCMS.java

```diff
@@ -60,7 +60,7 @@ public static SCM parseSCM(StaplerRequest req, AbstractProject target) throws Fo
         if (scm == null) {
             scm = new NullSCM(); // JENKINS-36043 workaround for AbstractMultiBranchProject.submit
         }
-        scm.getDescriptor().generation++;
+        scm.getDescriptor().generation.incrementAndGet();
```

Comment:
no need to get it here

Reply:
I didn't want to change the behavior there and ++ is exaclty incrementAndGet(). And there is no method which just increments the value. So either getAndIncrement() or the chosen one.

Reply:
OK, thanks